packages/zwave-js/src/lib/commandclass/HumidityControlSetpointCC.ts

```diff
@@ -0,0 +1,939 @@
+					this.driver.configManager,
+					setpointCaps.minValueScale,
+				);
+				const maxValueUnit = getSetpointUnit(
+					this.driver.configManager,
+					setpointCaps.maxValueScale,
+				);
+				const logMessage = `received capabilities for setpoint ${setpointName}:
+minimum value: ${setpointCaps.minValue} ${minValueUnit}
+maximum value: ${setpointCaps.maxValue} ${maxValueUnit}`;
+				this.driver.controllerLog.logNode(node.id, {
+					endpoint: this.endpointIndex,
+					message: logMessage,
+					direction: "inbound",
+				});
+			}
+		}
+
+		// Query the current value for all setpoint types
+		await this.refreshValues();
+
+		// Remember that the interview is complete
+		this.interviewComplete = true;
+	}
+
+	public async refreshValues(): Promise<void> {
+		const node = this.getNode()!;
+		const endpoint = this.getEndpoint()!;
+		const api = endpoint.commandClasses[
+			"Humidity Control Setpoint"
+		].withOptions({
+			priority: MessagePriority.NodeQuery,
+		});
+
+		const setpointTypes: HumidityControlSetpointType[] =
+			this.getValueDB().getValue(
+				getSupportedSetpointTypesValueID(this.endpointIndex),
+			) ?? [];
+
+		// Query each setpoint's current value
+		for (const type of setpointTypes) {
+			const setpointName = getEnumMemberName(
+				HumidityControlSetpointType,
+				type,
+			);
+			// Every time, query the current value
+			this.driver.controllerLog.logNode(node.id, {
+				endpoint: this.endpointIndex,
+				message: `querying current value of setpoint ${setpointName}...`,
+				direction: "outbound",
+			});
+			const setpoint = await api.get(type);
+			if (setpoint) {
+				const logMessage = `received current value of setpoint ${setpointName}: ${
+					setpoint.value
+				} ${setpoint.scale.unit ?? ""}`;
+				this.driver.controllerLog.logNode(node.id, {
+					endpoint: this.endpointIndex,
+					message: logMessage,
+					direction: "inbound",
+				});
+			}
+		}
+	}
+}
+
+interface HumidityControlSetpointCCSetOptions extends CCCommandOptions {
+	setpointType: HumidityControlSetpointType;
+	value: number;
+	scale: number;
+}
+
+@CCCommand(HumidityControlSetpointCommand.Set)
+export class HumidityControlSetpointCCSet extends HumidityControlSetpointCC {
+	public constructor(
+		driver: Driver,
+		options:
+			| CommandClassDeserializationOptions
+			| HumidityControlSetpointCCSetOptions,
+	) {
+		super(driver, options);
+		if (gotDeserializationOptions(options)) {
+			// TODO: Deserialize payload
+			throw new ZWaveError(
+				`${this.constructor.name}: deserialization not implemented`,
+				ZWaveErrorCodes.Deserialization_NotImplemented,
+			);
+		} else {
+			this.setpointType = options.setpointType;
+			this.value = options.value;
+			this.scale = options.scale;
+		}
+	}
+
+	public setpointType: HumidityControlSetpointType;
+	public value: number;
+	public scale: number;
+
+	public serialize(): Buffer {
+		const override =
+			this.getNodeUnsafe()?.deviceConfig?.compat?.overrideFloatEncoding;
```

Comment:
I'd prefer to leave this out for now unless there are devices which need this. This is only used for some Heatit thermostats so far.

Reply:
removed from 2 places